[SSPROD-8709] Use repository metadata in kernel crawler, take 2

gnosek opened the pull request:
Summary of changes since the previous attempt:

+ Skip some more debian/ubuntu kernels
+ Exclude `kernel-modules` CentOS packages
+ Handle recursion manually for sqlite < 3.8.0
+ Python 3 fixes
+ Fedora kernels can use `kernel-modules` RPMs too
+ Install python3 in the builder container

gnosek committed:
Fedora kernels can use `kernel-modules` RPMs too

gnosek committed:
Revert "Revert SSPROD-8709"

This reverts commit ca188ec3f141156c314e1738b297127d6df77e9d.

gnosek committed:
Python 3 fixes

gnosek committed:
[SSPROD-8709] Handle recursion manually for sqlite < 3.8.0

sqlite 3.7 does not support `WITH` recursive queries, so we have to
implement the recursive part in the script itself. The logic mostly
follows what a SQL-level recursive query would do:
- build a set of `pkgkey`s from the base case
- keep expanding the set through package dependencies
  until there are no new packages
- fetch the URLs to each of the found packages and return them all

gnosek committed:
Exclude `kernel-modules` CentOS packages

We don't need them for the build

gnosek committed:
Skip some more debian/ubuntu kernels

gnosek commented:
Verified this at least runs on the probe-builder box

deepskyblue86 reviewed (dismissed):
Looks good (I compared it to f1bb0e4) even if I have no clue about the query :)

iurly commented on kernel-crawler.py:
```diff
@@ -624,22 +624,50 @@ def build_base_query(cls, version=''):
+                cursor.execute(query, list(pkgkeys))
+                for row in cursor.fetchall():
+                    repo_db.append((ver, row[0]))
+            return repo_db
```
Is there any particular reason for having to maintain two different versions of the code which should be doing exactly the same thing?
I mean, is the recursive query significantly more efficient than the one performed by hand?
If not, I'd rather just always use the manually unfolded version (and perhaps keep the cleaner one commented as a future reference for when we'll want to deprecate older sqlite3 versions).
Just my two cents.

gnosek replied:
I commented the recursive variant out. Maybe we'll be able to switch to it in the future

gnosek committed:
Comment out the recursive SQLite query

It is useful as documentation and eventually using when we phase out
SQLite < 3.8 support but for now, just use one implementation
instead of two

iurly commented:
@gnosek I guess this PR is now obsolete, can we close it?
